autorest/azure/async.go

```diff
@@ -117,6 +119,45 @@ func (f Future) GetPollingDelay() (time.Duration, bool) {
 	return d, true
 }
 
+// WaitForCompletion will return when one of the following conditions is met: the long
+// running operation has completed, the provided context is cancelled, or the client's
+// polling duration has been exceeded.  It will retry failed polling attempts based on
+// the retry value defined in the client up to the maximum retry attempts.
+func (f Future) WaitForCompletion(ctx context.Context, client autorest.Client) error {
+	start := time.Now()
+	done, err := f.Done(client)
+	for attempts := 0; !done && attempts < client.RetryAttempts; done, err = f.Done(client) {
```

Comment:
> attempts < client.RetryAttempts [](start = 28, length = 32)

Is returning an error in case `attempts > RetryAttempts` useful?

Reply:
```err``` should already be non-nil in this case.  Or do you mean wrap it in an outer error?

Reply:
Yes I meant adding extra info on already done retries.

Reply:
Fixed.